pkg/virt-handler/options.go

```diff
@@ -54,23 +54,34 @@ func virtualMachineOptionsWithNetworkConfig(
 	capabilities *api.Capabilities,
 	disksInfo map[string]*containerdisk.DiskInfo,
 	clusterConfig *virtconfig.ClusterConfig,
+	vmiSpecIfaces []v1.Interface,
 ) (*cmdv1.VirtualMachineOptions, error) {
 	options := virtualMachineOptions(smbios, period, preallocatedVolumes, capabilities, disksInfo, clusterConfig)
 	if clusterConfig == nil {
 		return options, nil
 	}
+
+	domainAttachmentByPluginName := make(map[string]string)
 	networkBindings := clusterConfig.GetNetworkBindings()
-	if networkBindings == nil {
-		return options, nil
+	if networkBindings != nil {
+		for name, binding := range networkBindings {
+			if binding.DomainAttachment != "" {
+				domainAttachmentByPluginName[name] = string(binding.DomainAttachment)
+			}
+		}
 	}
-	var domainAttachments map[string]string
-	for name, binding := range networkBindings {
-		if binding.DomainAttachment != "" {
-			domainAttachments[name] = string(binding.DomainAttachment)
+
+	domainAttachmentByInterfaceName := make(map[string]string)
+	for _, iface := range vmiSpecIfaces {
+		if iface.Masquerade != nil || iface.Bridge != nil || iface.Macvtap != nil {
```

Comment:
I find it problematic to see this logic present so deep here. If possible, I would prefer the logic to exist under a network package, so we will not need to touch this place in the future if the network logic changes.

Same goes for the other processing above.

Reply:
Extracted the method to `"kubevirt.io/kubevirt/pkg/network/domainspec".
TODO: add unit tests to `domainspec.DomainAttachmentByInterfaceName`.